src/main/java/org/folio/rest/impl/InvoiceHelper.java

```diff
@@ -237,8 +237,18 @@ private boolean isUserDoesNotHaveDesiredPermission(AcqDesiredPermissions acqPerm
       });
   }
 
+  /**
+   * Delete Invoice
+   * 1. Get invoice by id
+   * 2. Verify if user has permission to delete Invoice based on acquisitions units
+   * 3. If user has permission to delete then delete invoiceLine
+   * @param id invoiceLine id to be deleted
+   */
   public CompletableFuture<Void> deleteInvoice(String id) {
-    return handleDeleteRequest(resourceByIdPath(INVOICES, id, lang), httpClient, ctx, okapiHeaders, logger);
+    return getInvoiceRecord(id)
+    .thenCompose(invoice -> protectionHelper.isOperationRestricted(invoice.getAcqUnitIds(), ProtectedOperationType.DELETE)
+        .thenApply(aVoid -> invoice))
+    .thenAccept(v -> handleDeleteRequest(resourceByIdPath(INVOICES, id, lang), httpClient, ctx, okapiHeaders, logger));
```

Comment:
If `getInvoiceRecord` and `protectionHelper.isOperationRestricted` are completed successfully, the user will always get 204 code regardless of the result of the `handleDeleteRequest` method. Please use `thenCompose` to include the result in the future chain.